src/board.rs

```diff
@@ -131,14 +144,23 @@ impl Board {
             }
         }
 
-        for y in 0..self.height() {
-            for x in 0..self.width() {
+        let rows = self.height();
+        let cols = self.width();
+        let squares = (0..rows).flat_map(|y| (0..cols).zip(std::iter::repeat(y)));
+
+        squares
+            .flat_map(|(x, y)| {
                 let c = Coordinate { x, y };
                 if !attatched.contains(&c) {
-                    self.clear(c);
+                    self.clear(c).map(|detail| Change {
+                        detail,
+                        change: ChangeAction::Truncated,
+                    })
+                } else {
+                    None
```

Comment:
Shouldn't this give us an iterator of length width*height containing mostly `None`? Or is `None` not collected?

Reply:
`flat_map` removes `None` values, and turns `Some(T)` values into `T` :)

Reply:
So you just wind up with an iterator over the `T` values